### ISSUE TITLE
Use of char* should generate compile time error

### ISSUE THREAD

**Issue by davidchappelle** [msgpack/msgpack-c#167]
Msgpack-c will implicitly convert char* to bool when creating objects. This can introduce hard to find bugs. Also, when used in conjunction with zones, cryptic compiler errors are generated complaining about the usage of char*. I feel that it would be reasonable to reject the usage of char* in the object api at compile time if possible.

**Comment by redboltz** [comment]
Sorry for my late reply. I considered the issue. 
See the following code:
```C++
#include <msgpack.hpp>
#include <iostream>
#include <cstring>
#include <cstdlib>

void print(char const* p) {
    std::cout << p << std::endl;
}

int main() {
    std::size_t const size = 256;
    char *buf = static_cast<char*>(std::malloc(size));
    if (!buf) return -1;
    std::strncpy(buf, "ABC", size);
    print(buf);
    msgpack::zone z;
    msgpack::object o(buf, z);
    std::cout << o << std::endl;
    free(buf);
}
```

The function print can work with char* even though the parameter is char const*. It's a natural behavior. However, msgpack doesn't work. I believe that the expected behavior is that buf dispathes to const char*. It is the same behavior as print() above.

So I sent PR as follows:
https://github.com/msgpack/msgpack-c/pull/206

**Comment by nobu-k** [comment]
Fixed via #206. `char*` is nowpassed to the overloaded function receiving `const char*` so that it won't generate a compile error.
